src/mem/localalloc.h

```diff
@@ -509,9 +511,20 @@ namespace snmalloc
         std::cout << "Large deallocation: " << size
                   << " chunk sizeclass: " << slab_sizeclass << std::endl;
 #endif
-        ChunkRecord* slab_record =
-          reinterpret_cast<ChunkRecord*>(entry.get_metaslab());
-        slab_record->chunk = CapPtr<void, CBChunk>(p);
+
+        /*
+         * TODO: There's nothing to say that this dealloc() is paired with the
+         * most recent allocation of this object, but we can at least check that
+         * it hasn't become freed, resized, or used as slab in the interim.
+         */
+        MetaEntry& entry_mut =
+          SharedStateHandle::get_meta_data_mut(address_cast(p));
+        ChunkRecord* slab_record = entry_mut.claim_for_large_dealloc(sizeclass);
+        if (unlikely(slab_record == nullptr))
+        {
+          return;
```

Comment:
Should this be an error that we report.  At least in the `SNMALLOC_CHECK_CLIENT` mode.  Also, can we switch off checking?  I would really like any enforcement of correct usage to be under `SNMALLOC_CHECK_CLIENT`, then we can get a true base-line for manual memory management checking costs.